common/src/math.rs

```diff
@@ -1,34 +1,107 @@
+///   Such points can be treated as oriented planes.
+///
+/// If the absolute value of this Minkowski inner product is 1, it is
+/// normalized, and equations involving such a vector tend to be simpler, much
+/// like with unit vectors.
+///
+/// Note that the simplest way to represent directions/velocities/normals at a
+/// point in hyperbolic space is with a vector whose Minkowski inner product
+/// with that point is 0. Such a vector will be tangent to the hyperboloid model
+/// at that associated point, so it can naturally represent movement along the
+/// hyperboloid in that direction.
+///
+/// As a general rule, when working with such vectors, it is highly recommended
+/// to avoid dot products and related operations such as vector magnitude, as
+/// these operations are meaningless in Minkowski space and are not preserved by
+/// isometries.
 #[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq)]
 #[repr(C)]
 pub struct MVector<N: Scalar>(na::Vector4<N>);
 
+/// An `MVector` with the constraint that the Minkowski inner product between
+/// the vector and itself is -1. Such a vector can be used to represent a point
+/// in hyperbolic space.
 #[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq)]
-#[repr(C)]
-pub struct MIsometry<N: Scalar>(na::Matrix4<N>);
+#[repr(transparent)]
+pub struct MUnitPointVector<N: Scalar>(MVector<N>);
+
+impl<N: Scalar> Deref for MUnitPointVector<N> {
+    type Target = MVector<N>;
 
-impl<N: RealField> From<na::Unit<na::Vector3<N>>> for MVector<N> {
-    fn from(value: na::Unit<na::Vector3<N>>) -> Self {
-        Self(value.into_inner().push(na::zero()))
+    fn deref(&self) -> &Self::Target {
+        &self.0
     }
 }
 
+/// An `MVector` with the constraint that the Minkowski inner product between
+/// the vector and itself is 1. Such a vector can be used to represent a
+/// direction in hyperbolic space.
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq)]
+#[repr(transparent)]
+pub struct MUnitDirectionVector<N: Scalar>(MVector<N>);
+
+impl<N: Scalar> Deref for MUnitDirectionVector<N> {
+    type Target = MVector<N>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// A stack-allocated, column-major, 4x4 square matrix in Minkowski space that
+/// preserves the Minkowski inner product. Such matrices are useful for
+/// computations in the hyperboloid model of hyperbolic space. Note that the
+/// last coordinate, not the first coordinate, is treated as the special "time"
+/// coordinate.
+///
+/// To ensure that this matrix indeed represents an isometry in Minkowski space,
+/// a few invariants are preserved:
+/// - The Minkowski inner product between any two distinct columns is 0.
+/// - The Minkowski inner product of a column with itself is 1 for the first
+///   three columns, and -1 for the last column.
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq)]
+#[repr(C)]
```

Comment:
Why `repr(C)` here and `repr(transparent)` above? Either is probably fine, but we should be consistent.

Reply:
Avoidance of touching older code. I'll go ahead and change them all to `transparent`.